ERCS/erc-7754.md

```diff
@@ -0,0 +1,256 @@
+
+1. The user's browser verifies the domain certificate and displays appropriate warnings if overtaken
+2. The DNS record of the dapp hosts a TXT field pointing to a URL where a JSON manifest is hosted
+   - This file SHOULD be at a well known address such as `https://domain.com/.well-known/twit.json`
+3. The config file contains an array of objects of the form `{ id, alg, publicKey }`
+4. For signed requests, the dapp first securely signs the payload with a private key, for example by submitting a request to its backend
+5. The original payload, signature, and public key id are sent to the wallet via the `wallet_signedRequest` RPC method
+6. The wallet verifies the signature before processing the request normally
+
+### Wallet integration
+
+#### Key discovery
+
+Attested public keys are necessary for the chain of trust to be established.
+Since this is traditionally done via DNS certificates, we propose the addition of a DNS record containing the public keys.
+This is similar to DKIM, but the use of a manifest file provides more flexibility for future improvements, as well as support for multiple algorithm and key pairs.
+
+Similarly to standard JWT practices, the wallet could eagerly cache dapp keys.
+However, in the absence of a revocation mechanism, a compromised key could still be used until caches have expired.
+To mitigate this, wallets SHOULD NOT cache dapp public keys for more than 2 hours.
+This practice establishes a relatively short vulnerability window, and manageable overhead for both wallet and dapp maintainers.
+
+Example DNS record for `my-crypto-dapp.org`:
+
+```txt
+...
+TXT: TWIT=/.well-known/twit.json
+```
+
+Example TWIT manifest at `https://my-crypto-dapp.org/twit.json`:
+
+```json
+{
+  "publicKeys": [
+    { "id": "1", "alg": "ECDSA", "publicKey": "0xaf34..." },
+    { "id": "2", "alg": "RSA-PSS", "publicKey": "0x98ab..." }
+  ]
+}
+```
+
+Dapps SHOULD only rely on algorithms available via [SubtleCrypto](https://www.w3.org/TR/2017/REC-WebCryptoAPI-20170126/), since they are present in every browser.
+
+#### Manifest schema
+
+We propose a simple and extensible schema:
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "TWIT manifest",
+  "type": "object",
+  "properties": {
+    "publicKeys": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": { "type": "string" },
+          "alg": { "type": "string" },
+          "publicKey": { "type": "string" }
+        }
+      }
+    }
+  }
+}
+```
+
+#### RPC method
+
+The parameters of `wallet_signedRequest` are specified by this TypeScript interface:
+
+```typescript
+type RequestPayload<Params> = { method: string; params: Params };
+
+type SignedRequestParameters<Params> = [
+  requestPayload: RequestPayload<Params>,
+  signature: `0x${string}`, // TODO: Is this the format we want?
```

Comment:
If you use HTML-style comments (outside the code fence), the linter will make sure you deal with them before moving into Review (eg. `<!-- TODO -->`.)